### PR TITLE
Add basic pre-training validation

### PR DESCRIPTION
**Pull request by Erin007**
[AI-25](https://codedotorg.atlassian.net/browse/AI-25)

validation

### CODE REVIEW

File: src/UIComponents/TrainModel.jsx [hunk 1]
```diff
@@ -34,53 +50,94 @@ class TrainModel extends Component {
...
-                going to look for patterns in these features:{" "}
-                {this.props.selectedFeatures.join(", ")} that might help predict
-                the values of the label: {this.props.labelColumn}.
-              </p>
-              <button type="button" onClick={this.onClickTrainModel}>
-                Train model
-              </button>
-              {this.state.showAccuracy && (
+        <h2>Are you ready to train the model?</h2>
+        {readyToTrainErrors.noFeature && (
```
Comment (breville) [inline src/UIComponents/TrainModel.jsx:57]: I wonder if we could create a JS data structure which maps these errors to their error strings?

File: src/redux.js [hunk 2]
```diff
@@ -354,6 +354,32 @@ export function getAccuracy(state) {
   return ((numCorrect / numPredictedLabels) * 100).toFixed(2);
 }
 
+export function readyToTrain(state) {
+  return !Object.values(readyToTrainErrors(state)).includes(true);
+}
+
+export function readyToTrainErrors(state) {
+  const readyToTrainErrors = {};
+  readyToTrainErrors["noFeature"] = state.selectedFeatures.length === 0;
```
Comment (breville) [inline src/redux.js:363]: Can we also just say `readyToTrainErrors.noFeature = `?

**Review by breville — APPROVED** [review]

File: src/UIComponents/TrainModel.jsx (hunk 1 above)
Comment (Erin007) [reply src/UIComponents/TrainModel.jsx:57]: oooh that's better! I was wondering about something like that I when I realized in the UI I had to know the specific keys to use to access the info in `readyToTrainErrors`.  I'll update.

Comment (breville) [reply src/UIComponents/TrainModel.jsx:57]: Longer term, this kind of conditional feedback reminds me of the "guide" we built for Oceans at https://github.com/code-dot-org/ml-activities/blob/main/src/oceans/models/guide.js.

**Pull request merged**